### PR TITLE
Pim 4186

### PR DESCRIPTION
**Pull request by sumbobyboys**

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 1]
```diff
@@ -109,12 +108,7 @@ define([
             },
             setValues: function (values) {
                 if (values.length === 0) {
-                    values.push(AttributeManager.getValue(
-                        [],
-                        this.attribute,
-                        this.context.locale,
-                        this.context.scope
-                    ));
+                    alert('value array is empty');
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: values ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/manager/attribute-manager.js [hunk 2]
```diff
@@ -90,36 +86,32 @@ define([
                             channel.code
                         );
 
-                        if (!_.findWhere(
-                            values,
-                            {scope: newValue.scope, locale: newValue.locale}
-                        )) {
+                        if (!newValue) {
+                            newValue = this.generateValue(attribute, locale, scope)
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/manager/attribute-manager.js]: scope -> channel ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/state.js [hunk 3]
```diff
@@ -106,6 +106,12 @@ define(
                 return false;
             },
             hasModelChanged: function () {
+                if (this.state !== JSON.stringify(this.getRoot().model.toJSON())) {
+                    /*global console: true */
+                    console.log(this.state);
+                    console.log(JSON.stringify(this.getRoot().model.toJSON()));
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/state.js]: Don't forget to remove during merge.

**Comment by sumbobyboys** [comment]
GTM

**Comment by jmleroux** [comment]
GTM

**Pull request merged**